projects/admin-core/src/lib/application/state/application.selectors.ts

```diff
@@ -203,3 +208,39 @@ export const selectServiceLayerTreeForApplication = createSelector(
 export const selectStylingConfig = createSelector(selectDraftApplication, application => application?.styling);
 
 export const selectFilterGroups = createSelector(selectDraftApplication, application => application?.settings?.filterGroups || []);
+
+export const selectFilterableLayersForApplication = createSelector(
+  selectAppLayerNodesForSelectedApplication,
+  selectGeoServiceLayers,
+  selectGeoServices,
+  selectFeatureTypes,
+  (appLayersNodes, geoServiceLayers, geoServices, featureTypes): GeoServiceLayerInApplicationModel[] => {
+    const geoServiceLayerMap = ApplicationTreeHelper.getLayerMap(geoServiceLayers);
+    return appLayersNodes
+      .filter(layer => ApplicationModelHelper.isLayerTreeNode(layer))
+      .map(layerNode => ({
+        geoServiceLayer: geoServiceLayerMap.get(ApplicationTreeHelper.getLayerMapKey(layerNode.layerName, layerNode.serviceId)),
+        appLayerId: layerNode.id,
+      }))
+      .filter(({ geoServiceLayer, appLayerId: _appLayerId }) => {
+        if (!geoServiceLayer) {
+          return false;
+        }
+        const geoService = geoServices.find(service => service.id === geoServiceLayer.serviceId);
+        const isGeoServer = geoService?.settings?.serverType === AdminServerType.GEOSERVER
+          || ((geoService?.settings?.serverType === AdminServerType.AUTO && geoService.resolvedServerType === AdminServerType.GEOSERVER));
+        if (!geoServiceLayer.layerSettings?.featureType) {
```

Comment:
Why not add `!isGeoServer` here as well, you can skip the code below if its not geoserver